zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -1003,6 +1003,45 @@ public static List<ACL> fixupACL(String path, List<Id> authInfo, List<ACL> acls)
             throw new KeeperException.InvalidACLException(path);
         }
         List<ACL> rv = new ArrayList<>();
+
+        // Overwrite the acl list for users (except super user) when znode group acl feature is on;
```

Comment:
When znode group acl is on? This is different from my understanding...

This is my understanding: the "user-provided acl" is overriden if `isX509ClientIdAsAclEnabled` is true. Have we decided to make changes to this?

Reply:
Your understanding is partially correct. What I meant by saying "when znode group acl feature is on" is "the auth provider is X509ZnodeGroupAclProvider, and isX509ClientIdAsAclEnabled is true". I think these two are both necessary.

Reply:
Then we are on the same page. Can we make sure we say the following:

> "the auth provider is X509ZnodeGroupAclProvider, and isX509ClientIdAsAclEnabled is true"

> "when znode group acl feature is on"
The statement above could mean something different. I think for most readers, it'd just mean "the auth provider is X509ZnodeGroupAclProvider".